orttraining/orttraining/python/training/_binary.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
If this is specific to hdf5, name it as hdf5_checkpoint.py or something like that. If it is generic, call it _checkpoint_storage.py

`binary` is kind of too deceiving. Can mean many things not related to checkopint